spec/features/registration_spec.rb

```diff
@@ -0,0 +1,34 @@
+feature 'registration' do
+  scenario 'a user can sign up' do
+    visit '/users/new'
+    fill_in('email', with: 'test@example.com')
+    fill_in('password', with: 'password123')
+    fill_in('name', with: 'testname')
+    fill_in('username', with: 'testusername')
+    click_button('Submit')
+
+    expect(page).to have_content "Welcome, testname"
+  end
+
+  scenario 'a user can not sign up with same email twice' do
+    visit '/users/new'
+    fill_in('email', with: 'test@example.com')
+    fill_in('password', with: 'password123')
+    fill_in('name', with: 'testname2')
+    fill_in('username', with: 'testusername2')
+    click_button('Submit')
+
+    expect(page).to_not have_content "Welcome, testname2"
```

Comment:
It would be nice here and in following feature test to also expect page to have the appropriate error message, e.g. 'Username or e-mail already in use" - I see these errors are already in place and working in the actual app, but it might be good to include them in the tests as well.

Reply:
good point!